st2client/tests/unit/test_auth.py

```diff
@@ -39,6 +41,108 @@
 }
 
 
+class TestLogin(base.BaseCLITestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(TestLogin, self).__init__(*args, **kwargs)
+        self.parser = argparse.ArgumentParser()
+        self.parser.add_argument('-t', '--token', dest='token')
+        self.parser.add_argument('--api-key', dest='api_key')
+        self.shell = shell.Shell()
+
+    def setUp(self):
+        super(TestLogin, self).setUp()
+
+    def tearDown(self):
+        super(TestLogin, self).tearDown()
+
+    @mock.patch.object(
+        requests, 'post',
+        mock.MagicMock(return_value=base.FakeResponse(json.dumps({}), 200, 'OK')))
+    @mock.patch("st2client.commands.auth.ConfigParser")
+    @mock.patch("st2client.commands.auth.open")
+    @mock.patch("st2client.commands.auth.BaseCLIApp")
+    @mock.patch("st2client.commands.auth.getpass")
+    def test_login(self, mock_gp, mock_cli, mock_open, mock_cfg):
+
+        test_cases = [
+            {
+                "args": [
+                    '--config', '/tmp/st2config', 'login', 'st2admin', '--password', 'Password1!'
+                ],
+                "expected_username": 'st2admin'
+            },
+            {
+                "args": ['login', 'st2admin'],
+                "expected_username": 'st2admin'
+            },
+            {
+                "args": ['login', 'st2admin', '--password', 'Password1!', '-w'],
+                "expected_username": 'st2admin'
+            },
+            {
+                "args": ['--config', '/tmp/st2config', 'login', 'st2admin', '-w'],
+                "expected_username": 'st2admin'
+            }
+        ]
```

Comment:
Okay I split up these tests and added a function `_login_common_assertions` to save a little bit of linecount on common things between the unit tests. Committed in d45b9af